python/lsst/ts/pmd/component.py

```diff
@@ -186,23 +187,30 @@ async def get_slots_position(self):
                 split_reply = reply.decode().split(":")
                 position[i] = float(split_reply[-1])
             else:
-                position[i] = math.nan
-                # Try to recover multiplexer
-                self.log.error(
-                    f"Sensor {name} is not reading. {self.read_error_count} continuous read errors reported"
-                )
-
-                if self.read_error_count == 2 and self.multiplexer_recovery():
-                    self.log.info("Multiplexer recovered")
-                    self.read_error_count = 0
-                else:
-                    self.read_error_count += 1
+                while not self.read_error_count >= 3 and not self.mutliplexer_recovered:
+                    position[i] = math.nan
+                    self.read_error_count = 1
+                    # Try to recover multiplexer
                     self.log.error(
-                        "Multiplexer recovery failed "
-                        f"{self.read_error_count} times"
-                        " continuously"
+                        (
+                            f"Sensor {name} is not reading."
+                            f"{self.read_error_count} continuous read errors reported"
+                        )
                     )
 
+                    if await self.multiplexer_recovery():
+                        self.mutliplexer_recovered = True
+                        self.log.info("Multiplexer recovered")
+                        self.read_error_count = 0
+                        break
+                    else:
+                        self.read_error_count += 1
+                        self.log.error(
+                            "Multiplexer recovery failed "
+                            f"{self.read_error_count} times"
+                            " continuously"
+                        )
+
```

Comment:
I don't really understand what you are doing. What I see is:
* For each channel you send a command and read a reply.
* If the reply is b"\r" (which is just the byte string that is "\n" -- please remove the comment "unsure what b"\r" is...") then you await self.multiplexer_recovery().
* But you do not try to read the reply for that name again. So that position is guaranteed to be "nan".
 
So why are you recovering? Why not read the position if you successfully recover?